SteamProject/MySteam/Steam.cpp

```diff
@@ -0,0 +1,261 @@
+#include "Steam.h"
+#include <iostream>
+
+
+void FSteam::RunApp()
+{
+	while (true)
+	{
+		int UserOption;
+		std::cout << "------ Welcome to Steam! ------" << std::endl;
+		std::cout << "\t1 - Add New game \n\t2 - Categories Menu... \n\t3 - Show Games... \n\t4 - Exit \n" << ">>";
+		std::cin >> UserOption;
+
+		switch (UserOption)
+		{
+		case AddGame:
+			OpenAddGameMenu();
+			break;
+
+		case ShowCategories:
+			OpenCategoriesMenu();
+			break;
+
+		case ShowGames:
+			OpenShowGamesMenu();
+			break;
+
+		case Exit:
+			exit(1);
+			break;
+
+		default:
+			std::cin.clear();
+			std::cin.ignore(100, '\n');
+			break;
+		}
+
+		system("cls");
+	}
+}
+
+
+int FSteam::ValidateInput(const std::string Question)
+{
+	bool bRepeat;
+	do
+	{
+		std::cout << Question << " >>";
+		int Input;
+		std::cin >> Input;
+
+		if (std::cin.fail())
+		{
+			bRepeat = true;
+			std::cin.clear();
+			std::cin.ignore(100, '\n');
+			system("cls");
+		}
+		else
+		{
+			bRepeat = false;
+			return Input;
+		}
+
+	} while (bRepeat);
+}
+
+FGame FSteam::AskForNewGame()
+{
+	bool bRepeat;
+	do
+	{
+		system("cls");
+
+		std::cout << "Enter the game name >>";
+		std::string NewGameName;
+		std::cin >> NewGameName;
```

Comment:
What happens if the game has more than one word? Same as the studio name.

Reply:
I change the std::cin for a std::getLine